src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -216,17 +218,35 @@ def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper
     return task_list_file, len(genome_ids_chunks)
 
 
-def _create_batch_script(job_dir, task_list_file, n_jobs):
+def _cal_node_num(tool, n_jobs):
+    """
+    Calculate the number of nodes required for the task
+    """
+
+    tool_exe_time = TASK_META[tool]['exe_time']
+    jobs_per_node = NODE_TIME_LIMIT * 60 // tool_exe_time
+
+    num_nodes = min(math.ceil(n_jobs / jobs_per_node), MAX_NODE_NUM)
+
+    return num_nodes
+
+
+def _create_batch_script(job_dir, task_list_file, n_jobs, tool):
     """
     Create the batch script (submit_taskfarmer.sl) for job submission
     """
+
+    node_num = _cal_node_num(tool, n_jobs)
+
     batch_script = f'''#!/bin/sh
 
-#SBATCH -N {n_jobs + 1} -c 64
+#SBATCH -N {node_num + 1} -c 256
```

Comment:
Remind me what `-c` does?

Reply:
cpu num available for the workers.

Reply:
Ok, that's what I assumed. I thought we said a ways back that there was diminishing returns to adding CPUs to GTDB (at least), and it might make sense to run multiple instances of the tool on the same node at the same time.

Reply:
Do you still have questions on this? I am not sure what you are asking.

Reply:
I'm pretty sure I remember that the original benchmarking you did showed that there was a point where it wasn't very useful to throw more cores at GTDBTK and the time gains were minimal. IIRC we discussed that rather than assigning more cores per GTDBTK process, we'd get better throughput by running multiple GTDBTK processes on the same machine, splitting the cores between them.

This PR is quadrupling the number of cores for the GTDBTK process (unless I don't understand what's going here, which is def possible) which seems to contradict the above.

Reply:
oh I see what you mean now. Each Perlmutter node has 256 cores. By doing `-c 256`,  I want to take all cores available. (I need to review this with Shane to be sure). Running multiple tasks on the same node is done via the arguments by `compute_genome_attri.py`. Right now, it was hardcoded as `threads=256, program_threads=256` in https://github.com/kbase/collections/blob/main/src/loaders/jobs/taskfarmer/task_generator.py#L179-L180
Meaning, we want to use 256 cores and each task also takes 256 cores (1 task per node). Once this is fully tested, I want to introduce `program_threads` argument, maybe some variable configured by tools. If we set it to 128, there will be 2 tasks running on the same node. But `-c` in the batch script doesn't affect how we want to schedule tasks per node.

Reply:
Ok, is there a ticket that covers changing the system to run > 1 task per node somewhere?

Reply:
I really don't think we need a ticket for every single thing especially since we are developing something not well-defined. But https://kbase-jira.atlassian.net/browse/RE2022-113

Reply:
I don't want to lose track of this stuff, and unless we make a ticket we will *shrug*